app/src/main/java/com/example/sportsmates/place/domain/PlacesUseCase.kt

```diff
@@ -10,14 +10,14 @@ class PlacesUseCase(
 ) {
     suspend fun getRelatedPlaces(): List<Place?> {
         val userLocation = userPreferences.user?.city
-        val places = placesRepository.getPlaces()
+        val places = placesRepository.getAllPlaces()
         val filteredPlaces =
             places.filter { place ->
                 place?.city == userLocation
             }.map { place ->
-            place?.mainImage = placesRepository.getPlaceMainImage(place?.placeId)
+                place?.mainImage = placesRepository.getPlaceMainImage(place?.placeId)
```

Comment:
use async and await to make a parallel network calls , this will reduce the loading time .
this what i mean -> 
.map { place ->
                coroutineScope {
                    async {
                        place?.mainImage = placesRepository.getPlaceMainImage(place?.placeId)
                        place
                    }
                }
        }
  return filteredPlaces.awaitAll() .